string payload
uint32 message_type